src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2504,25 +2528,27 @@ public bool HasBoxingConversion(TypeSymbol source, TypeSymbol destination, ref H
 
             if (source.IsRestrictedType())
             {
-                return false;
+                return ConversionKind.NoConversion;
             }
 
             if (destination.Kind == SymbolKind.DynamicType)
             {
-                return !source.IsPointerType();
+                return !source.IsPointerType()
+                    ? ConversionKind.ValueTypeBoxing
+                    : ConversionKind.NoConversion;
```

Comment:
similar issue.  "source" coudl be a type-parameter here that we're boxing because it's assigned to a dynamic type.  However, the boxing is happening due to us knowing this is a value type.  SO i think ValueTypeBoxing is appropriate.

Thoughts?